### PR TITLE
termios: Support FreeBSD

### PR DESCRIPTION
**Pull request by patmaddox**
This sets some variables necessary to support Tailscale's built-in sshd server on FreeBSD (https://github.com/tailscale/tailscale/pull/6155).

I could use some help finishing this off. Current issues that I think are worth addressing:

- [x] `pkg/termios/termios_freebsd.go` is a `cp` of `pkg/termios/termios_darwin.go`, with the addition of the `go:build freebsd` / `+build freebsd` lines.
    - Addressed by renaming to `termios_bsd.go` and conditionally building.
- [x] `pkg/termios/var_freebsd.go` has a lot of duplication with `pkg/termios/var_unix.go` as well.
    - Addressed by moving some syscalls out of `var_unix.go` and into `var_linux.go` and `var_darwin.go`.

From what I can tell, CI doesn't currently verify FreeBSD.

#2421 says:

> Remove cirrus ci: It was only used for verifying freebsd buildability. We are going to use github actions for that instead (#2419)

GitHub actions doesn't officially support FreeBSD as far as I know. [The method I'm aware of](https://github.com/marketplace/actions/freebsd-vm) requires you to run a FreeBSD VM inside of a MacOS instance. That's pretty janky.

Cirrus CI runs directly on FreeBSD machines, so it should probably be re-instated for FreeBSD verification.

I am willing to assist (re-)configuring FreeBSD CI on Cirrus.

### CODE REVIEW

**Review by rminnich — APPROVED** [review]
This sure looks good to me. If you can make cirrus CI
work it would be nice to have freebsd CI back. Help wanted :-)

**Review by rminnich — APPROVED** [review]
from what I've seen, ttyio is always going to be a maze, I think this is about as good a job as can be done. Thanks again.

File: pkg/termios/var_freebsd.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build freebsd
```
Comment (rminnich) [inline pkg/termios/var_freebsd.go:5]: note that with this file name, I doubt you need the build tag.

**Review by rminnich — APPROVED** [review]

**Comment by patmaddox** [comment]
Removed unnecessary build line from `var_freebsd.go`, and rebased off of main.

**Comment by patmaddox** [comment]
> This sure looks good to me. If you can make cirrus CI work it would be nice to have freebsd CI back. Help wanted :-)

Cool! Happy to do it. Would you like to see it as part of this PR, or a separate one?

**Review by rminnich — APPROVED** [review]

**Pull request merged**